# Titanic Data

1. Create a model that includes only age, fare, and pclass. Does this model perform better than your baseline?

2. Include sex in your model as well. Note that you'll need to encode or create a dummy variable of this feature before including it in a model.

3. Try out other combinations of features and models.

4. Use you best 3 models to predict and evaluate on your validate sample.

5. Choose you best model from the validation performation, and evaluate it on the test dataset. How do the performance metrics compare to validate? to train?

### Bonus questions:
- Bonus1 How do different strategies for handling the missing values in the age column affect model performance?

- Bonus2: How do different strategies for encoding sex affect model performance?

- Bonus3: scikit-learn's LogisticRegression classifier is actually applying a regularization penalty to the coefficients by default. This penalty causes the magnitude of the coefficients in the resulting model to be smaller than they otherwise would be. This value can be modified with the C hyper parameter. Small values of C correspond to a larger penalty, and large values of C correspond to a smaller penalty. Try out the following values for C and note how the coefficients and the model's performance on both the dataset it was trained on and on the validate split are affected.
   C =.01,.1,1,10,100,1000
- Bonus Bonus: how does scaling the data interact with your choice of C?

In [2]:
import numpy as np
import pandas as pd
import math
import matplotlib.pyplot as plt 
import acquire as acq
import prepare as prep

from sklearn.metrics import classification_report, confusion_matrix, plot_confusion_matrix
from sklearn.linear_model import LogisticRegression

In [2]:
titanic=acq.titanic_data()
titanic=prep.prep_titanic(titanic)
titanic.head()


passenger_id  survived  pclass     sex   age  sibsp  parch     fare  \
0             0         0       3    male  22.0      1      0   7.2500   
1             1         1       1  female  38.0      1      0  71.2833   
2             2         1       3  female  26.0      0      0   7.9250   
3             3         1       1  female  35.0      1      0  53.1000   
4             4         0       3    male  35.0      0      0   8.0500   

   embark_town  sex_male  embark_town_Queenstown  embark_town_Southampton  
0  Southampton         1                       0                        1  
1    Cherbourg         0                       0                        0  
2  Southampton         0                       0                        1  
3  Southampton         0                       0                        1  
4  Southampton         1                       0                        1

1. Create a model that includes only age, fare, and pclass. Does this model perform better than your baseline?

baseline is %61.61 and model1 is 68.35. Model1 does perform better than the baseline

2. Include sex in your model as well. Note that you'll need to encode or create a dummy variable of this feature before including it in a model.

In [3]:
baseline_accuracy=329/(329+205)
baseline_accuracy*100

61.61048689138576

In [4]:
df1=titanic[['age','fare','pclass','survived']]
df1=df1.fillna(0)

In [5]:
titanic.sex=(titanic.sex=="male")*1

In [6]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       891 non-null    float64
 1   fare      891 non-null    float64
 2   pclass    891 non-null    int64  
 3   survived  891 non-null    int64  
dtypes: float64(2), int64(2)
memory usage: 28.0 KB


In [7]:
train,validate,test=prep.split_data(df1,'survived')

In [8]:
x_train=train.drop(columns=['survived'])
y_train=train.survived
x_validate=validate.drop(columns=['survived'])
y_validate=validate.survived
x_test=test.drop(columns=['survived'])
y_test=test.survived

In [9]:
logit=LogisticRegression()
logit.fit(x_train,y_train)
logit.score(x_train,y_train)*100

68.35205992509363

In [10]:
df2=titanic[['age','fare','pclass','sex','survived']]
df2=df2.fillna(0)

In [11]:
train2,validate2,test2=prep.split_data(df2,'survived')

In [12]:
x_train2=train2.drop(columns=['survived'])
y_train2=train2.survived
x_validate2=validate2.drop(columns=['survived'])
y_validate2=validate2.survived
x_test2=test2.drop(columns=['survived'])
y_test2=test2.survived

In [13]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

77.90262172284645

3. Try out other combinations of features and models.

## Model #3 

In [14]:
df3=titanic[['pclass','sibsp','survived']]
df3=df3.fillna(0)

In [15]:
train3,validate3,test3=prep.split_data(df3,'survived')

In [16]:
x_train3=train3.drop(columns=['survived'])
y_train3=train3.survived
x_validate3=validate3.drop(columns=['survived'])
y_validate3=validate3.survived
x_test3=test3.drop(columns=['survived'])
y_test3=test3.survived

In [17]:
logit3=LogisticRegression()
logit3.fit(x_train3,y_train3)
logit3.score(x_train3,y_train3)*100

67.22846441947566

## Model #4

In [18]:
df4=titanic[['sex_male','parch','survived']]
df4=df4.fillna(0)

In [19]:
train4,validate4,test4=prep.split_data(df4,'survived')

In [20]:
x_train4=train4.drop(columns=['survived'])
y_train4=train4.survived
x_validate4=validate4.drop(columns=['survived'])
y_validate4=validate4.survived
x_test4=test4.drop(columns=['survived'])
y_test4=test4.survived

In [21]:
logit4=LogisticRegression()
logit4.fit(x_train4,y_train4)
logit4.score(x_train4,y_train4)*100

79.21348314606742

## Model #5

In [22]:
df5=titanic[['fare','embark_town_Southampton','survived']]
df5=df5.fillna(0)

In [23]:
train5,validate5,test5=prep.split_data(df5,'survived')

In [24]:
x_train5=train5.drop(columns=['survived'])
y_train5=train5.survived
x_validate5=validate5.drop(columns=['survived'])
y_validate5=validate5.survived
x_test5=test5.drop(columns=['survived'])
y_test5=test5.survived

In [25]:
logit5=LogisticRegression()
logit5.fit(x_train5,y_train5)
logit5.score(x_train5,y_train5)*100

66.10486891385767

## Model #6

In [26]:
df6=titanic[['age','pclass','embark_town_Southampton','survived']]
df6=df6.fillna(0)

In [27]:
train6,validate6,test6=prep.split_data(df6,'survived')

In [28]:
x_train6=train6.drop(columns=['survived'])
y_train6=train6.survived
x_validate6=validate6.drop(columns=['survived'])
y_validate6=validate6.survived
x_test6=test6.drop(columns=['survived'])
y_test6=test6.survived

In [29]:
logit6=LogisticRegression()
logit6.fit(x_train6,y_train6)
logit6.score(x_train6,y_train6)*100

67.22846441947566

In [30]:
baseline_accuracy*100,logit.score(x_train,y_train)*100, logit2.score(x_train2,y_train2)*100, logit3.score(x_train3,y_train3)*100,logit4.score(x_train4,y_train4)*100,logit5.score(x_train5,y_train5)*100,logit6.score(x_train6,y_train6)*100

(61.61048689138576,
 68.35205992509363,
 77.90262172284645,
 67.22846441947566,
 79.21348314606742,
 66.10486891385767,
 67.22846441947566)

4. Use you best 3 models to predict and evaluate on your validate sample.
 - my top three models are:
  model 2, model 3, model 4

### Model 2 predict and evaluate on validate sample:
#### Train: 77.90%

In [31]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       891 non-null    float64
 1   fare      891 non-null    float64
 2   pclass    891 non-null    int64  
 3   sex       891 non-null    int64  
 4   survived  891 non-null    int64  
dtypes: float64(2), int64(3)
memory usage: 34.9 KB


In [33]:
logit2.score(x_validate2,y_validate2)*100

77.52808988764045

In [41]:
y_pred=logit2.predict(x_validate2)

In [43]:
confusion_matrix(y_validate2,y_pred)

array([[90, 20],
       [20, 48]])

In [57]:
report2 = classification_report(y_validate2, y_pred, output_dict=True)

In [58]:
cr2 = pd.DataFrame(report2).transpose().round(2)
cr2

precision  recall  f1-score  support
0                  0.69    0.86      0.77   110.00
1                  0.62    0.37      0.46    68.00
accuracy           0.67    0.67      0.67     0.67
macro avg          0.66    0.62      0.61   178.00
weighted avg       0.66    0.67      0.65   178.00

### Model 3 predict and evaluate on validate sample:
#### Train: 67.23%

In [52]:
logit3.score(x_validate3,y_validate3)*100

67.41573033707866

In [59]:
y_pred=logit3.predict(x_validate3)

In [60]:
confusion_matrix(y_validate3,y_pred)

array([[95, 15],
       [43, 25]])

In [61]:
report3 = classification_report(y_validate3, y_pred, output_dict=True)
cr3 = pd.DataFrame(report3).transpose().round(2)
cr3

precision  recall  f1-score  support
0                  0.69    0.86      0.77   110.00
1                  0.62    0.37      0.46    68.00
accuracy           0.67    0.67      0.67     0.67
macro avg          0.66    0.62      0.61   178.00
weighted avg       0.66    0.67      0.65   178.00

### Model 4 predict and evaluate on validate sample:
#### Train: 79.21%

In [62]:
logit4.score(x_validate4,y_validate4)*100

77.52808988764045

In [73]:
y_pred=logit4.predict(x_validate4)

In [65]:
confusion_matrix(y_validate4,y_pred)

array([[91, 19],
       [21, 47]])

In [74]:
report_v4 = classification_report(y_validate4, y_pred, output_dict=True)
cr_v4 = pd.DataFrame(report_v4).transpose().round(2)
cr_v4

precision  recall  f1-score  support
0                  0.81    0.83      0.82   110.00
1                  0.71    0.69      0.70    68.00
accuracy           0.78    0.78      0.78     0.78
macro avg          0.76    0.76      0.76   178.00
weighted avg       0.77    0.78      0.77   178.00

5. Choose you best model from the validation performation, and evaluate it on the test dataset. How do the performance metrics compare to validate? to train?

### Model 4 did best on validate sample:
#### Train: 79.21 %
#### Validate: 77.53%
#### Test: 78.21%

In [67]:
logit4.score(x_test4,y_test4)*100

78.2122905027933

In [70]:
y_pred=logit4.predict(x_test4)

In [71]:
report_t4 = classification_report(y_test4, y_pred, output_dict=True)
cr_t4 = pd.DataFrame(report_t4).transpose().round(2)
cr_t4

precision  recall  f1-score  support
0                  0.80    0.85      0.83   110.00
1                  0.74    0.67      0.70    69.00
accuracy           0.78    0.78      0.78     0.78
macro avg          0.77    0.76      0.77   179.00
weighted avg       0.78    0.78      0.78   179.00

In [114]:
l_score4=logit4.score(x_train4,y_train4)*100
v_score4=logit4.score(x_validate4,y_validate4)*100
t_score4=logit4.score(x_test4,y_test4)*100
baseline=baseline_accuracy*100

In [115]:
metrics=[]
output = {
        "train_score": l_score4,
        "validate_score":v_score4 ,
        "test_score": t_score4,
        
        
    }
    
metrics.append(output)
metrics=pd.DataFrame(metrics)

In [116]:
metrics

train_score  validate_score  test_score
0    79.213483        77.52809   78.212291

### Bonus1:  How do different strategies for handling the missing values in the age column affect model performance?
##### 1. using mean on age column, model 4: the percentage is 70.60
##### 2. using the mode percentage stayed the same as using the mean.
##### 3. using LOCF percentage dropped about 1-2%

In [177]:
# USING MEAN for age
titanic=acq.titanic_data()
titanic=prep.prep_titanic(titanic)
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   passenger_id             891 non-null    int64  
 1   survived                 891 non-null    int64  
 2   pclass                   891 non-null    int64  
 3   sex                      891 non-null    object 
 4   age                      714 non-null    float64
 5   sibsp                    891 non-null    int64  
 6   parch                    891 non-null    int64  
 7   fare                     891 non-null    float64
 8   embark_town              891 non-null    object 
 9   sex_male                 891 non-null    uint8  
 10  embark_town_Queenstown   891 non-null    uint8  
 11  embark_town_Southampton  891 non-null    uint8  
dtypes: float64(2), int64(5), object(2), uint8(3)
memory usage: 65.4+ KB


In [179]:
titanic.age.mean()

29.69911764705882

In [180]:
df1=titanic[['age','fare','pclass','survived']]
df1.age=titanic.age.fillna(29.70)
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       891 non-null    float64
 1   fare      891 non-null    float64
 2   pclass    891 non-null    int64  
 3   survived  891 non-null    int64  
dtypes: float64(2), int64(2)
memory usage: 28.0 KB


/var/folders/wd/5_z1tvw94pv9x2ld8p_0dl2r0000gn/T/ipykernel_18778/2276208686.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1.age=titanic.age.fillna(29.70)


In [181]:
train,validate,test=prep.split_data(df1,'survived')
x_train=train.drop(columns=['survived'])
y_train=train.survived
x_validate=validate.drop(columns=['survived'])
y_validate=validate.survived
x_test=test.drop(columns=['survived'])
y_test=test.survived

In [182]:
logit=LogisticRegression()
logit.fit(x_train,y_train)
logit.score(x_train,y_train)*100

70.59925093632958

In [183]:
logit=LogisticRegression()
logit.fit(x_train,y_train)
logit.score(x_validate,y_validate)*100

69.66292134831461

In [184]:
logit=LogisticRegression()
logit.fit(x_train,y_train)
logit.score(x_test,y_test)*100

71.50837988826815

In [186]:
#USING MODE
titanic=acq.titanic_data()
titanic=prep.prep_titanic(titanic)
titanic.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   passenger_id             891 non-null    int64  
 1   survived                 891 non-null    int64  
 2   pclass                   891 non-null    int64  
 3   sex                      891 non-null    object 
 4   age                      714 non-null    float64
 5   sibsp                    891 non-null    int64  
 6   parch                    891 non-null    int64  
 7   fare                     891 non-null    float64
 8   embark_town              891 non-null    object 
 9   sex_male                 891 non-null    uint8  
 10  embark_town_Queenstown   891 non-null    uint8  
 11  embark_town_Southampton  891 non-null    uint8  
dtypes: float64(2), int64(5), object(2), uint8(3)
memory usage: 65.4+ KB


In [189]:
titanic.age.mode()

0    24.0
Name: age, dtype: float64

In [194]:
df2.age.fillna(24)
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       891 non-null    float64
 1   fare      891 non-null    float64
 2   pclass    891 non-null    int64  
 3   survived  891 non-null    int64  
dtypes: float64(2), int64(2)
memory usage: 28.0 KB


In [195]:
train2,validate2,test2=prep.split_data(df2,'survived')

In [196]:
x_train2=train2.drop(columns=['survived'])
y_train2=train2.survived
x_validate2=validate2.drop(columns=['survived'])
y_validate2=validate2.survived
x_test2=test2.drop(columns=['survived'])
y_test2=test2.survived

In [197]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

70.4119850187266

In [198]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

69.10112359550563

In [199]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

71.50837988826815

In [212]:
# USIN LOCF
titanic=acq.titanic_data()
titanic=prep.prep_titanic(titanic)
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   passenger_id             891 non-null    int64  
 1   survived                 891 non-null    int64  
 2   pclass                   891 non-null    int64  
 3   sex                      891 non-null    object 
 4   age                      714 non-null    float64
 5   sibsp                    891 non-null    int64  
 6   parch                    891 non-null    int64  
 7   fare                     891 non-null    float64
 8   embark_town              891 non-null    object 
 9   sex_male                 891 non-null    uint8  
 10  embark_town_Queenstown   891 non-null    uint8  
 11  embark_town_Southampton  891 non-null    uint8  
dtypes: float64(2), int64(5), object(2), uint8(3)
memory usage: 65.4+ KB


In [213]:
df3=titanic[['age','fare','pclass','survived']]
df3["age"] = df3["age"].fillna(method='ffill')

/var/folders/wd/5_z1tvw94pv9x2ld8p_0dl2r0000gn/T/ipykernel_18778/3914239768.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3["age"] = df3["age"].fillna(method='ffill')


In [219]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       891 non-null    float64
 1   fare      891 non-null    float64
 2   pclass    891 non-null    int64  
 3   survived  891 non-null    int64  
dtypes: float64(2), int64(2)
memory usage: 28.0 KB


In [214]:
train3,validate3,test3=prep.split_data(df3,'survived')

In [215]:
x_train3=train3.drop(columns=['survived'])
y_train3=train3.survived
x_validate3=validate3.drop(columns=['survived'])
y_validate3=validate3.survived
x_test3=test3.drop(columns=['survived'])
y_test3=test3.survived

In [216]:
logit3=LogisticRegression()
logit3.fit(x_train3,y_train3)
logit3.score(x_train3,y_train3)*100

69.47565543071161

In [217]:
logit3=LogisticRegression()
logit3.fit(x_train3,y_train3)
logit3.score(x_validate3,y_validate3)*100

69.10112359550563

In [218]:
logit3=LogisticRegression()
logit3.fit(x_train3,y_train3)
logit3.score(x_test3,y_test3)*100

72.62569832402235

## Bonus 2: How do different strategies for encoding sex affect model performance?

is_female: 79.1
is_male: 78.1

In [224]:
titanic=acq.titanic_data()
titanic=prep.prep_titanic(titanic)
titanic.head()

passenger_id  survived  pclass     sex   age  sibsp  parch     fare  \
0             0         0       3    male  22.0      1      0   7.2500   
1             1         1       1  female  38.0      1      0  71.2833   
2             2         1       3  female  26.0      0      0   7.9250   
3             3         1       1  female  35.0      1      0  53.1000   
4             4         0       3    male  35.0      0      0   8.0500   

   embark_town  sex_male  embark_town_Queenstown  embark_town_Southampton  
0  Southampton         1                       0                        1  
1    Cherbourg         0                       0                        0  
2  Southampton         0                       0                        1  
3  Southampton         0                       0                        1  
4  Southampton         1                       0                        1

In [225]:
#FEMALE
titanic.sex=(titanic.sex=="female")*1

In [226]:
titanic.sex.value_counts()

0    577
1    314
Name: sex, dtype: int64

In [230]:
df2=titanic[['age','fare','pclass','sex','survived']]
df2.age=df2.age.fillna(29.70)

/var/folders/wd/5_z1tvw94pv9x2ld8p_0dl2r0000gn/T/ipykernel_18778/2422814527.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2.age=df2.age.fillna(29.70)


In [232]:
train2,validate2,test2=prep.split_data(df2,'survived')

In [233]:
x_train2=train2.drop(columns=['survived'])
y_train2=train2.survived
x_validate2=validate2.drop(columns=['survived'])
y_validate2=validate2.survived
x_test2=test2.drop(columns=['survived'])
y_test2=test2.survived

In [234]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

80.52434456928839

In [235]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

78.08988764044943

In [236]:
logit2=LogisticRegression()
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

79.3296089385475

In [237]:
titanic=acq.titanic_data()
titanic=prep.prep_titanic(titanic)
titanic.head()

passenger_id  survived  pclass     sex   age  sibsp  parch     fare  \
0             0         0       3    male  22.0      1      0   7.2500   
1             1         1       1  female  38.0      1      0  71.2833   
2             2         1       3  female  26.0      0      0   7.9250   
3             3         1       1  female  35.0      1      0  53.1000   
4             4         0       3    male  35.0      0      0   8.0500   

   embark_town  sex_male  embark_town_Queenstown  embark_town_Southampton  
0  Southampton         1                       0                        1  
1    Cherbourg         0                       0                        0  
2  Southampton         0                       0                        1  
3  Southampton         0                       0                        1  
4  Southampton         1                       0                        1

In [238]:
#MALE
titanic.sex=(titanic.sex=="male")*1

In [239]:
df3=titanic[['age','fare','pclass','sex','survived']]
df3.age=df3.age.fillna(29.70)

/var/folders/wd/5_z1tvw94pv9x2ld8p_0dl2r0000gn/T/ipykernel_18778/332253079.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3.age=df3.age.fillna(29.70)


In [240]:
train3,validate3,test3=prep.split_data(df3,'survived')

In [241]:
x_train3=train3.drop(columns=['survived'])
y_train3=train3.survived
x_validate3=validate3.drop(columns=['survived'])
y_validate3=validate3.survived
x_test3=test3.drop(columns=['survived'])
y_test3=test3.survived

In [242]:
logit3=LogisticRegression()
logit3.fit(x_train3,y_train3)
logit3.score(x_validate3,y_validate3)*100

78.08988764044943

## Bonus 3:Try out the following values for C and note how the coefficients and the model's performance on both the dataset it was trained on and on the validate split are affected.
 - C=.01,.1,1,10,100,1000

In [243]:
logit2=LogisticRegression(C=.01)
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

70.59925093632958

In [244]:
logit2=LogisticRegression(C=.01)
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

71.91011235955057

In [245]:
logit2=LogisticRegression(C=.01)
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

76.53631284916202

In [246]:
logit2=LogisticRegression(C=.1)
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

78.27715355805243

In [248]:
logit2=LogisticRegression(C=.1)
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

79.21348314606742

In [249]:
logit2=LogisticRegression(C=.1)
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

78.77094972067039

In [250]:
logit2=LogisticRegression(C=1)
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

80.52434456928839

In [251]:
logit2=LogisticRegression(C=1)
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

78.08988764044943

In [252]:
logit2=LogisticRegression(C=1)
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

79.3296089385475

In [253]:
logit2=LogisticRegression(C=10)
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

80.71161048689139

In [254]:
logit2=LogisticRegression(C=10)
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

77.52808988764045

In [255]:
logit2=LogisticRegression(C=10)
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

79.3296089385475

In [256]:
logit2=LogisticRegression(C=100)
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

80.52434456928839

In [257]:
logit2=LogisticRegression(C=100)
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

77.52808988764045

In [258]:
logit2=LogisticRegression(C=100)
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

79.3296089385475

In [259]:
logit2=LogisticRegression(C=1000)
logit2.fit(x_train2,y_train2)
logit2.score(x_train2,y_train2)*100

80.52434456928839

In [260]:
logit2=LogisticRegression(C=1000)
logit2.fit(x_train2,y_train2)
logit2.score(x_validate2,y_validate2)*100

77.52808988764045

In [261]:
logit2=LogisticRegression(C=1000)
logit2.fit(x_train2,y_train2)
logit2.score(x_test2,y_test2)*100

79.3296089385475